BackfillPowerBI.yml

```diff
@@ -0,0 +1,58 @@
+parameters:
+- name: DeleteAllOrganizationData
+  displayName: Delete all data in Power BI Simeon Data source
+  default: false
+  type: boolean
+
+- name: BackfillHours
+  displayName: Number of hours to include in backfill
+  default: 72
+  type: number
+
+name: $(Date:yyyyMMdd)$(Rev:.r)
+trigger: none
+pool:
+  vmImage: windows-latest
+resources:
+  repositories:
+  - repository: Reporting
+    type: github
+    name: simeoncloud/reporting
+    ref: refs/heads/feature/uploadToPowerBI
+    endpoint: simeoncloud
+jobs:
+- job: Job
+  timeoutInMinutes: 60
+  steps:
+  - checkout: Reporting
+
+  - pwsh: 
+      (Get-ChildItem -Filter 'Invoke-UploadPlanToPowerBI.ps1' -Recurse)[0].FullName | Import-Module -Force;
+      Connect-PowerBIAsServicePrincipal -AppId $env:AppId -AppSecret $env:Secret -Tenant $env:Tenant;
+      Clear-PowerBITable;
+    condition: and(succeeded(), ${{ parameters.DeleteAllOrganizationData }})
+    name: ClearPowerBITable
+    displayName: Clear all data from PowerBI
+    env:
+      AppId: $(AppId)
+      Secret: $(Secret)
+      Tenant: $(Tenant)
+
+  - pwsh: |
+      try {
+          (Get-ChildItem -Filter 'Invoke-UploadPlanToPowerBI.ps1' -Recurse)[0].FullName | Import-Module -Force
+          Connect-PowerBIAsServicePrincipal -AppId $env:AppId -AppSecret $env:Secret -Tenant $env:Tenant
+          Invoke-SimeonCloudReportingBackfill -OrganizationUri "$(System.CollectionUri)" -PersonalAccessToken $env:PersonalAccessToken -BackfillHours $env:BackfillHours
+      } catch {
+          Write-Error "There was an error"
+          Write-Error -ErrorRecord $_ -EA Continue
+          Write-Error -ErrorRecord (Resolve-PowerBIError -Last) -EA Continue
+      }
+    name: Backfill
+    displayName: Backfill PowerBI
+    env:
+      PersonalAccessToken: $(System.AccessToken)
```

Comment:
Just use inline in the script - the reason being for consistency - you already use System.CollectionUri in-line in the script

Reply:
Don't I need to keep this one since its secret?

Reply:
yes